crates/holochain/src/conductor/api/api_external/app_interface.rs

```diff
@@ -79,6 +79,15 @@ impl AppInterfaceApi {
                     .get_app_info(&installed_app_id)
                     .await?,
             )),
+            AppRequest::AgentInfo { dna_hash } => {
```

Comment:
Nice, this is a good breakdown here. Most of the work should happen in the conductor and then this just does the conversion to strings that is specific to the app interface.

If we don't maintain this boundary then using the conductor as a library requires copying code.